src/wallet/wallet.cpp

```diff
@@ -272,6 +272,8 @@ void CWallet::AvailableCoinsForStaking(vector<COutput>& vCoins, unsigned int nSp
                 }
         }
     }
+    
+    std::sort(vCoins.begin(), vCoins.end(), sortByCoinAgeDescending());
```

Comment:
I would add here `if (GetBoolArg("-stakingsortbycoinage",false))`

Reply:
I agree with this flag check

@dantudor can you add this flag check, then I'll approve PR